test/integration/node/lifecycle_test.go

```diff
@@ -152,7 +145,7 @@ func TestEvictionForNoExecuteTaintAddedByUser(t *testing.T) {
 			for index := range nodes {
 				nodes[index], err = cs.CoreV1().Nodes().Create(testCtx.Ctx, nodes[index], metav1.CreateOptions{})
 				if err != nil {
-					t.Errorf("Failed to create node, err: %v", err)
+					t.Fatalf("Failed to create node, err: %v", err)
```

Comment:
it's better if we do not conflate unrelated changes on the same commit

Reply:
I think replacing the Errorf with Fatalf for setup makes the test simpler (which is the goal of the PR).

Reply:
yeah, but think that those are 2 different tests, if we somehow figure out after merge that we broke something, we have to revert the commit, and we want to revert only the code that break something,
In this case, you should have one commit per test, we somehow organize commits per topic to facilitate reviews, but keeping a clean git history helps the code maintenance , is not a big deal in this case, but hygiene always help

Reply:
I get the point about keeping commits isolated and clean. However, I think in this case
case it makes sense to apply the same changes to both tests in the file (`TestTaintBasedEvictions` and `TestEvictionForNoExecuteTaintAddedByUser`):
- move `nodeIndex` next to `nodeCount`
- add a comment to `nodeIndex`
- using Fatalf instead of Errorf 

Applying them to only one of the tests would make another reader puzzled it seems.

Reply:
I've split the changes into two separate commits within the PR. Hope it makes sense.

Reply:
ah ok, then I misread it, sorry for the obvious comment then :smile:

Reply:
no worries, thanks